Fix deleting a value from an existing select product attribute while it's in use (Postgresql issue only)

coldic3 opened the pull request:
| Q               | A
|-----------------|-----
| Branch?         | 1.14
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes https://github.com/Sylius/Sylius/issues/17704
| License         | MIT

coldic3 committed:
[Behat] Add scenario for deleting a value from an existing selected product attribute while it's in use

coldic3 committed:
Fix ProductAttributeValueRepository::findByJsonChoiceKey with CAST function

coldic3 commented:
I use the cast function from the `oro/doctrine-extensions` package, but I'm open to changing it if you prefer.

GSadee commented:
> I use the cast function from the `oro/doctrine-extensions` package, but I'm open to changing it if you prefer.

Hi Kevin! Forgive the lack of reply 😅.  I am wondering about this new dependency, mainly because we are planning to add support to doctrine/orm 3, and it doesn't support this version so far. I don't know if it wouldn't be better to implement it here, like other methods. What do you think? 🤔

coldic3 commented:
@GSadee Good point, I'll try to replace this dependency with a custom CAST function implementation :woman_dancing:

coldic3 committed:
Replace the oro/doctrine-extensions dependency with a custom Cast Doctrine ORM function

coldic3 commented:
@GSadee Done, I've replaced the dependency with a custom CAST function. But please be aware that the function I implemented is much more simple that the one from `oro/doctrine-extensions` and does not cover all possible castings. I've focused only on `CAST(something AS text)`.
If someone uses it in some other way. I'm not sure about the result.

Another option is to add `CAST_TO_TEXT(something)` so then we are sure this always works. WDYT?

GSadee commented:
It took a while 😅, but thank you Kevin! 🎉
